### PR TITLE
Refactor usage of Pydantic aliases

### PR DESCRIPTION
**Pull request by disrupted**

### CODE REVIEW

File: tests/component_handlers/schema_handler/test_schema_handler.py [hunk 1]
```diff
@@ -149,8 +149,8 @@ def test_should_log_info_when_submit_schemas_that_not_exists_and_dry_run_true(
     )
     topic_config = TopicConfig(
         type=OutputTopicTypes.OUTPUT,
-        keySchema=None,
-        valueSchema="com.bakdata.kpops.test.SchemaHandlerTest",
+        key_schema=None,
```
Comment (philipp94831) [inline tests/component_handlers/schema_handler/test_schema_handler.py:152]: Is this breaking our existing configs?

Comment (disrupted) [reply tests/component_handlers/schema_handler/test_schema_handler.py:152]: no, they use aliases for the camel case version

File: kpops/components/base_components/models/to_section.py [hunk 2]
```diff
@@ -28,6 +28,7 @@ class TopicConfig(BaseModel):
 
     class Config(BaseConfig):
         extra = Extra.forbid
+        allow_population_by_field_name = True
```
Comment (raminqaf) [inline kpops/components/base_components/models/to_section.py:31]: Does this mean that we use the field name (like `shema_registry_url`) in the `pipeline.yaml`?

Comment (disrupted) [reply kpops/components/base_components/models/to_section.py:31]: yes, both snake case and camel case version are supported, although snake case should be preferred

**Review by raminqaf — APPROVED** [review]

**Pull request merged**